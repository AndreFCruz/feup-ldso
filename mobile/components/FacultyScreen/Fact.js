import React from 'react';
import {StyleSheet} from 'react-native';
import {View, Text} from 'native-base';
import PropTypes from 'prop-types';
import facultyStyles from '../../constants/SpecificStyles';
import {widthPercentageToDP as wp} from 'react-native-responsive-screen';

class Fact extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {name, info} = this.props;
        
        let nr_display = null;
        let displayText = info;
        let matches = displayText.match(/^(\d+)\s?/); //this regex checks if displayText varible starts with a number
        if (matches != null) {
            displayText = displayText.replace(/^(\d+)\s/, ''); //this regex removes the initial number from displayText variable
            nr_display = <Text style={[styles.number, facultyStyles[name].factText]}>{matches[1]}</Text>;
        }
        return (
            <View style={[styles.fact, facultyStyles[name].factBorder]}>
                <View style={[styles.factContent, facultyStyles[name].factBackground]}>
                    {nr_display}
                    <Text style={[facultyStyles[name].factText, (matches != null ? styles.text : styles.centerText)]}>{displayText}</Text>
                </View>
            </View>
        );
    }
}

Fact.propTypes = {
    name: PropTypes.string,
    info: PropTypes.string,
};


export { Fact };


const styles = StyleSheet.create({
    fact: {
        flex:1,
        borderWidth: 3,
        margin: wp('0.65%'),
        borderRadius: 18,
        borderStyle: 'dotted',
    },
    factContent: {
        flex: 1,
        margin: wp('0.8%'),
        borderRadius: 13,
        elevation:15,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row',
    },
    number: {
        flex:1,
        textAlign:'right',
        fontSize: wp('3%'),
        paddingRight: wp('2%'),
    },  
    text: {
        flex:2,
        fontSize: wp('1.3%'),
        paddingRight: wp('2%'),
    },
    centerText: {
        flex:2,
        fontSize: wp('1.3%'),
        textAlign: 'center'
    }
});

