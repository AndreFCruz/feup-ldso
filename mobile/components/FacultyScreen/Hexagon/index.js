import GridGenerator from './GridGenerator';
import HexGrid from './HexGrid';
import HexUtils from './HexUtils';
import Layout from './Layout';
import HexPath from './HexPath';
import Pattern from './Pattern';

import Hexagon from './Hexagon/Hexagon';
import HexText from './Hexagon/HexText';

import Hex from './models/Hex';

export { GridGenerator, HexGrid, HexUtils, Layout, HexPath, Pattern, Hexagon, HexText, Hex };
