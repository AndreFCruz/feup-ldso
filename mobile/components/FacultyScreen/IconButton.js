import React from 'react';
import {View, Card, Icon, Button, Text} from 'native-base';
import specificStyles from '../../constants/SpecificStyles';

class IconButton extends React.Component {
    render() {
        return (
            <Card style={styles.card} transparent>
                <View style={styles.btnView}>
                    <Button style={[styles.linkBtn, specificStyles.mainMenuBtn.feup]} /*onPress={() => this.props.navigation.navigate('Videos', {name: name})}*/>
                        <Icon style={[styles.icon,specificStyles.mainMenuIcon.feup]} type="FontAwesome" name="film" />
                    </Button>
                </View>
                <View style={styles.subtitle}>
                    <Text style={styles.labelText}>Videos</Text>
                </View>
            </Card>
        );
    }
}

export { IconButton };

const styles = {

    card: {
        padding: 30,
       alignItems: 'center'
    },

    btnView: {
        flex:1,
        flexDirection: 'column',
    },

    subtitle: {
        flex:1,
        flexDirection: 'column',
        alignItems: 'center',
        marginTop:83,
    },


    labelText: {
        fontSize: 22
    },

    icon: {
        transform: [{
            rotate: '315deg'
        }]
    },

    linkBtn: {
        width: 90,
        height: 90,
        justifyContent: 'center',
        borderWidth: 1.2,
        borderRadius: 6,
        backgroundColor: 'white',
        transform: [{
            rotate: '45deg'
        }]
  
    }
}