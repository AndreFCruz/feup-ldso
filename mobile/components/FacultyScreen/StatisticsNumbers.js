import React from 'react';
import {View} from 'react-native';
import {Text, Title} from 'native-base';
import Dash from 'react-native-dash';

class StatisticsNumbers extends React.Component {
    render() {
        return (
            <View style={styles.main}>
                <View style={styles.section}>
                    <Text style={styles.numbers}>13</Text>
                    <Text style={styles.text}>Licenciaturas</Text>
                </View>
                <View style={styles.section}>
                    <Dash dashColor='white' dashGap={5} style={styles.dash} />
                </View>
                <View style={styles.section}>
                    <Text style={styles.numbers}>2221</Text>
                    <Text style={styles.text}>Estudantes</Text>
                </View>
            </View>
        );
    };
}

export { StatisticsNumbers };

const styles = {
    main: {
        padding:20,
        flex: 1,
        flexDirection: 'column',
        backgroundColor: 'black'
    },

    section: {
        padding: 3,
        flex:1,
        alignItems: 'center',
        justifyContent: 'center'
    },

    dash: {
        width:1,
        height:50,
        flexDirection:'column'
    },

    numbers: {
        fontFamily: 'Quicksand_regular',
        fontSize: 25,
        color: 'white',
        paddingBottom: 5
    },

    text: {
        color: 'white'
    }
}