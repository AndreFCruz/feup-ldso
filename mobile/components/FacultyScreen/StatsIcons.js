import React from 'react';
import {View} from 'react-native';
import {Text, Icon} from 'native-base';
import PropTypes from 'prop-types';
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';

class StatisticsNumbers extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {icon, iconsNmb,  percentage, text} = this.props;
        
        var icons = [];
        let colored = 0;

        for(let i = 0; i < iconsNmb; i++) {

            if(colored < (iconsNmb*percentage/100)) {
                icons.push(
                    <Icon key={i} style={styles.icon} type="FontAwesome" name={icon}/>
                );
                colored++;
            } else {
                icons.push(
                    <Icon key={i} style={styles.backgroundIcon} type="FontAwesome" name={icon}/>
                );
            }
        }    

        return (
            <View style={styles.main}>
                <View style={[styles.section, styles.justify]}>
                    {icons}
                </View>
                <View style={[styles.section, styles.percentageContent]}>
                    <Text style={styles.percentage}>{percentage}%</Text>
                </View>
                <View style={styles.section}>
                    <Text style={styles.description}>{text}</Text>
                </View>
            </View>
        );
    }
}

StatisticsNumbers.propTypes = {
    icon: PropTypes.string,
    iconsNmb: PropTypes.number,
    percentage: PropTypes.number,
    text: PropTypes.string
};

export default StatisticsNumbers ;

const styles = {
    main: {
        flex: 1,
        flexDirection: 'column',
        padding: wp('1%'),
        paddingBottom: hp('0.5%'),
        alignItems: 'center',
        justifyContent: 'center',
    },

    section: {
        flex:1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        flexWrap: 'wrap',
    },

    justify: {
        justifyContent: 'center'
    },

    backgroundIcon: {
        fontSize: hp('9%'),
        padding: hp('0.5%'),
        color: '#cccccc'
    },

    icon: {
        fontSize: hp('9%'),
        padding: hp('0.5%'),
        color: '#1c1c1c'
    },

    percentage: {
        fontFamily: 'Quicksand_regular',
        fontSize: hp('5%'),
        color: 'black',
    },

    percentageContent: {
        paddingTop: hp('23.5%'),
        marginBottom: hp('5%'),
    },

    
    description: {
        justifyContent: 'center',
        alignItems: 'center',
        fontFamily: 'Quicksand_regular',
        fontSize: wp('1.6%'),
        textAlign: 'center'

    }

};