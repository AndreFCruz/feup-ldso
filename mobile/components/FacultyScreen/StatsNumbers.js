import React from 'react';
import {View} from 'react-native';
import {Text } from 'native-base';
import PropTypes from 'prop-types';
import Dash from 'react-native-dash';

class StatisticsNumbers extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {course, students} = this.props;

        return (
            <View style={styles.main}>
                <View style={styles.section}>
                    <Text style={styles.numbers}>{course[0]}</Text>
                    <Text style={styles.text}>{course[1]}</Text>
                </View>
                <View style={styles.section}>
                    <Dash dashColor='white' dashGap={5} style={styles.dash} />
                </View>
                <View style={styles.section}>
                    <Text style={styles.numbers}>{students[0]}</Text>
                    <Text style={styles.text}>{students[1]}</Text>
                </View>
            </View>
        );
    }
}

StatisticsNumbers.propTypes = {
    course: PropTypes.array,
    students: PropTypes.array,
};

export default StatisticsNumbers ;

const styles = {
    main: {
        padding:20,
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#1c1c1c'
    },

    section: {
        padding: 3,
        flex:1,
        alignItems: 'center',
        justifyContent: 'center'
    },

    dash: {
        width:1,
        height:50,
        flexDirection:'column'
    },

    numbers: {
        fontFamily: 'Quicksand_regular',
        fontSize: 30,
        color: 'white',
        paddingBottom: 5
    },

    text: {
        color: 'white',
        fontFamily: 'Quicksand_bold',
        fontSize: 25
    }
};