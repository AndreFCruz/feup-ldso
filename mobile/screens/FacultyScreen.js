import React from 'react';
import { StyleSheet, Image, Dimensions, ScrollView, TouchableOpacity } from 'react-native';
import { Container, Content, Text, View, Icon } from 'native-base';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getIntroduction, setFaculty } from '../reducers/modules/facultyReducer';
import Statistics from '../components/FacultyScreen/StatsNumbers';
import IconButton from '../components/FacultyScreen/IconButton';


let scrollYPos = 0;

class FacultyScreen extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            screenHeight: Dimensions.get('window').height-80,      
            screenWidth: Dimensions.get('window').width,
        };
    }

    scrollToSecondPage = () => {
        scrollYPos = this.state.screenHeight * 1;
        this.scroller.scrollTo({x: 0, y: scrollYPos});
    };

    componentDidMount() {
        const faculty = this.props.navigation.getParam('faculty');

        this.props.setFaculty(faculty);
        this.props.getIntroduction(this.props.language, faculty);
    }

    componentDidUpdate(prevProps) {
        const { language, name } = this.props;

        if (prevProps.language != language)
            this.props.getIntroduction(language, name);
    }

    navigateFunction(label, name) {
        return () => this.props.navigation.navigate(label, {name: name});
    }

    render() {
        const { name, loading, language } = this.props;

        if (loading) {
            return (
                <Container style={styles.container}>
                    <Content contentContainerStyle={styles.content}>
                        <Text style={styles.text}>
                            { language == 'en'? 'Loading...' : 'Carregando...' }
                        </Text>
                    </Content>
                </Container>
            );
        }
        return (
            <ScrollView style={styles.container} ref={(scroller) => {this.scroller = scroller;}}>
                <Content contentContainerStyle={styles.content}>
                    <View style={styles.menu}>
                        <View style={styles.links}>
                            <View style={styles.linksRow}>
                                <IconButton name={name} icon="film" label="Videos" action={() => this.navigateFunction('Videos', name)}></IconButton>
                                <IconButton name={name} icon="globe" label="Social Projects" action={() => this.navigateFunction('SocialProjects', name)}></IconButton>
                            </View>
                            <View style={styles.linksRow}>
                                <IconButton name={name} icon="paper-plane" label="Future" action={() => this.navigateFunction('FutureProspects', name)}></IconButton>
                                <IconButton name={name} icon="map-marker" label="Localization" action={() => this.navigateFunction('Localization', name)}></IconButton>
                            </View>
                        </View>
                        <View style={styles.imageView}>
                            <Image style={styles.image} source={{uri: 'https://static.globalnoticias.pt/jn/image.aspx?brand=JN&type=generate&guid=151d9c4c-8a02-466b-95fe-1ae900791412&w=744&h=495&t=20180406133500'}}/>
                        </View>
                    </View>
                    <View style={styles.firstPageBottom}>
                        <View style={styles.statistics}>
                            <Statistics course={[ '13', 'Faculdades' ]} students={['2221', 'Estudantes']}></Statistics>
                            <Statistics course={[ '28', 'Mestrados' ]} students={['961', 'Estudantes']}></Statistics>
                            <Statistics course={[ '8', 'Doutoramentos' ]} students={['280', 'Estudantes']}></Statistics>
                            <Statistics course={[ '125', 'Cursos de Formação' ]} students={['1111', 'Formandos']}></Statistics>
                        </View>
                        <TouchableOpacity style={styles.goDownArrow} onPress={this.scrollToSecondPage}>
                            <Icon style={styles.goDownArrowIcon} type="FontAwesome" name="chevron-down" />
                        </TouchableOpacity>
                    </View>
                </Content>
                <Content contentContainerStyle={styles.content}>
                    <View style={styles.statsIcons}>
                    </View>
                    <View style={styles.hexagonsView}>
                    </View>
                </Content>
            </ScrollView>
        );
    }
}

FacultyScreen.propTypes = {
    name: PropTypes.string,
    loading: PropTypes.bool,
    language: PropTypes.string,
    navigation: PropTypes.object,
    setFaculty: PropTypes.func,
    getIntroduction: PropTypes.func
};

const styles = StyleSheet.create({
    container: {
        flex:1,
        flexDirection: 'column',
        backgroundColor: 'white',
        
    },

    content: {
        flexDirection: 'column',
        height: Dimensions.get('window').height-80,
        justifyContent: 'center',
        padding:0
    },

    menu: {
        flex:2,
        flexDirection: 'row',
    },

    firstPageBottom: {
        flex:1,
        flexDirection: 'column',
        backgroundColor: '#1c1c1c'
    },

    statistics: {
        flex: 5,
        flexDirection: 'row',
        justifyContent: 'space-between'
    },

    goDownArrow: {
        flex:1,
        alignItems: 'center'
    },

    goDownArrowIcon: {
        color: 'white'
    },

    h1: {
        padding: 30
    },

    image: {
        flex: 1,
        margin: 20,
        marginBottom: 37,
        resizeMode: 'contain',
        alignSelf: 'stretch'

    },

    imageView: {
        flex: 3,
    },

    text: {
        textAlign: 'justify',
        padding: 10
    },

    links: {
        flex: 2,
        flexDirection: 'column',
        justifyContent: 'space-between',
        paddingTop: 10
    },

    linksRow: {
        flex:1,
        flexDirection: 'row',
        justifyContent: 'space-around',
        alignItems: 'stretch'
    },

    statsIcons: {
        flex:9,
        flexDirection: 'column',
        backgroundColor: 'white'
    },

    hexagonsView: {
        flex:11,
        flexDirection: 'column',
        backgroundColor: '#1c1c1c',
        alignItems: 'center'
    }

});

const mapStateToProps = ({ faculty, language }) => ({
    name: faculty.name,
    loading: faculty.loading,
    intro: faculty.intro,
    language: language.selection
});

const mapDispatchToProps = {
    setFaculty,
    getIntroduction
};

export default connect(mapStateToProps, mapDispatchToProps)(FacultyScreen);
