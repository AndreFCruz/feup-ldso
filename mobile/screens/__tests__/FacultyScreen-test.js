import React from 'react';
import FacultyScreen from '../FacultyScreen';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const props = { navigation: {navigate: jest.fn(), name:'feup', getParam: jest.fn() } };

jest.mock('ScrollView', () => {
    const RealComponent = require.requireActual('ScrollView');
    const React = require('React');
    class ScrollView extends React.Component {
        constructor(props){
            super(props);
        }
        scrollTo = () => {

        }
  
        render() {
            return React.createElement('ScrollView', this.props, this.props.children);
        }
    }
    ScrollView.propTypes = RealComponent.propTypes;
    return ScrollView;
});


function setup(loading = false, facts = 5) {
    let otherFacts;
    if(facts == 2) {
        otherFacts = ['240 protocolos estágio com empresas', '30 contratos projetos prestação de serviços'];
    } else {
        otherFacts = [
            '240 protocolos estágio com empresas',
            '30 contratos projetos prestação de serviços',
            '17 protocolos com escolas de ensino básico e secundário',
            '30 trabalhos de tradução e revisão jurídica',
            '8 campanhas de angariação de bens',
        ];
    }
    const initialState = { 
        faculty: {
            loading: loading,
            name: 'feup',
            stats: {
                'bsc_students': 2221,
                'faculty': 'feup',
                'foreign_student_perc': 0.06,
                'msc_students': 961,
                'nr_bsc': 13,
                'nr_msc': 28,
                'nr_phd': 8,
                'nr_training_course': 125,
                'other_facts': otherFacts,
                'phd_students': 280,
                'research_perc': 0.9,
                'training_course_graduate': 111,
                'training_programs_perc': 0.18,
            }
        },
        language: {
            selection: 'en'
        }
    };

    const wrapper = shallow(<FacultyScreen {...props}/>,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: false,
            lifecycleExperimental: true
        });

    return wrapper;
}

describe('Rendering', () => {
    it('renders loading screen', () => {
        const wrapper = setup(true);

        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders screen with many facts', () => {
        const wrapper = setup();
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders screen with 2 facts', () => {
        const wrapper = setup(false, 2);
        expect(wrapper.dive()).toMatchSnapshot();
    });
});


