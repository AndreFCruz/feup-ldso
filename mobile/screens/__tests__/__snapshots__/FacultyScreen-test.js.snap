// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders loading screen 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Styled(Content)
    contentContainerStyle={
      Object {
        "flexDirection": "column",
        "height": 1254,
        "justifyContent": "center",
        "padding": 0,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "padding": 53.5,
          "textAlign": "justify",
        }
      }
    >
      Loading...
    </Styled(Text)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`Rendering renders screen with 2 facts 1`] = `
<ScrollView
  onMomentumScrollBegin={null}
  onScrollEndDrag={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Styled(Content)
    contentContainerStyle={
      Object {
        "flexDirection": "column",
        "height": 1254,
        "justifyContent": "center",
        "padding": 0,
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "flex": 3,
          "flexDirection": "row",
          "paddingBottom": 40,
          "paddingTop": 40,
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 3,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "flex": 3,
              "flexDirection": "column",
              "justifyContent": "space-around",
            }
          }
        >
          <IconButton
            action={[Function]}
            icon="videos"
            label="Videos"
            name="feup"
          />
          <IconButton
            action={[Function]}
            icon="socialProjects"
            label="Social Projects"
            name="feup"
          />
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Array [
              Object {
                "flex": 3,
                "flexDirection": "column",
                "justifyContent": "space-around",
              },
              Object {
                "flex": 2,
                "paddingBottom": "25%",
                "paddingTop": "17%",
              },
              Object {
                "alignItems": "flex-start",
              },
            ]
          }
        >
          <IconButton
            action={[Function]}
            icon="research"
            label="Research Centres"
            name="feup"
          />
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Object {
              "flex": 3,
              "flexDirection": "column",
              "justifyContent": "space-around",
            }
          }
        >
          <IconButton
            action={[Function]}
            icon="future"
            label="Future"
            name="feup"
          />
          <IconButton
            action={[Function]}
            icon="location"
            label="Location"
            name="feup"
          />
        </Styled(ViewNB)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 3,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginRight": 15,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "resizeMode": "cover",
            }
          }
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "backgroundColor": "#1c1c1c",
          "flex": 2,
          "flexDirection": "column",
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 15,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <StatisticsNumbers
          course={
            Array [
              13,
              "Bachelors
",
            ]
          }
          students={
            Array [
              2221,
              "Students",
            ]
          }
        />
        <StatisticsNumbers
          course={
            Array [
              28,
              "Masters
",
            ]
          }
          students={
            Array [
              961,
              "Students",
            ]
          }
        />
        <StatisticsNumbers
          course={
            Array [
              8,
              "PhD
",
            ]
          }
          students={
            Array [
              280,
              "Students",
            ]
          }
        />
        <StatisticsNumbers
          course={
            Array [
              125,
              "Open and continuing training courses
",
            ]
          }
          students={
            Array [
              111,
              "Students",
            ]
          }
        />
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-end",
            }
          }
        >
          <Styled(Icon)
            name="chevron-down"
            style={
              Object {
                "color": "white",
              }
            }
            type="FontAwesome"
          />
        </TouchableOpacity>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Content)>
  <Styled(Content)
    contentContainerStyle={
      Object {
        "flexDirection": "column",
        "height": 1254,
        "justifyContent": "center",
        "padding": 0,
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "flex": 9,
          "flexDirection": "column",
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-start",
            }
          }
        >
          <Styled(Icon)
            name="chevron-up"
            type="FontAwesome"
          />
        </TouchableOpacity>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "backgroundColor": "white",
            "flex": 15,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-evenly",
            "padding": 40,
            "paddingTop": 0,
          }
        }
      >
        <StatisticsNumbers
          icon="flask"
          iconsNmb={18}
          percentage={90}
          text="Teachers and Reseachers

"
        />
        <StatisticsNumbers
          icon="globe"
          iconsNmb={18}
          percentage={6}
          text="Foreign students in mobility programs

"
        />
        <StatisticsNumbers
          icon="book"
          iconsNmb={18}
          percentage={18}
          text="Foreign students enrolled to obtain a degree
"
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1c1c1c",
          "flex": 8,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <Styled(ViewNB)
        key="0"
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "flex": 1,
            "flexDirection": "column",
            "padding": "2%",
          }
        }
      >
        <Fact
          info="240 protocolos estágio com empresas"
          key="0"
          name="feup"
        />
      </Styled(ViewNB)>
      <Styled(ViewNB)
        key="1"
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "flex": 1,
            "flexDirection": "column",
            "padding": "2%",
          }
        }
      >
        <Fact
          info="30 contratos projetos prestação de serviços"
          key="0"
          name="feup"
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Content)>
</ScrollView>
`;

exports[`Rendering renders screen with many facts 1`] = `
<ScrollView
  onMomentumScrollBegin={null}
  onScrollEndDrag={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Styled(Content)
    contentContainerStyle={
      Object {
        "flexDirection": "column",
        "height": 1254,
        "justifyContent": "center",
        "padding": 0,
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "flex": 3,
          "flexDirection": "row",
          "paddingBottom": 40,
          "paddingTop": 40,
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 3,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "flex": 3,
              "flexDirection": "column",
              "justifyContent": "space-around",
            }
          }
        >
          <IconButton
            action={[Function]}
            icon="videos"
            label="Videos"
            name="feup"
          />
          <IconButton
            action={[Function]}
            icon="socialProjects"
            label="Social Projects"
            name="feup"
          />
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Array [
              Object {
                "flex": 3,
                "flexDirection": "column",
                "justifyContent": "space-around",
              },
              Object {
                "flex": 2,
                "paddingBottom": "25%",
                "paddingTop": "17%",
              },
              Object {
                "alignItems": "flex-start",
              },
            ]
          }
        >
          <IconButton
            action={[Function]}
            icon="research"
            label="Research Centres"
            name="feup"
          />
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Object {
              "flex": 3,
              "flexDirection": "column",
              "justifyContent": "space-around",
            }
          }
        >
          <IconButton
            action={[Function]}
            icon="future"
            label="Future"
            name="feup"
          />
          <IconButton
            action={[Function]}
            icon="location"
            label="Location"
            name="feup"
          />
        </Styled(ViewNB)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 3,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginRight": 15,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "resizeMode": "cover",
            }
          }
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "backgroundColor": "#1c1c1c",
          "flex": 2,
          "flexDirection": "column",
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 15,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <StatisticsNumbers
          course={
            Array [
              13,
              "Bachelors
",
            ]
          }
          students={
            Array [
              2221,
              "Students",
            ]
          }
        />
        <StatisticsNumbers
          course={
            Array [
              28,
              "Masters
",
            ]
          }
          students={
            Array [
              961,
              "Students",
            ]
          }
        />
        <StatisticsNumbers
          course={
            Array [
              8,
              "PhD
",
            ]
          }
          students={
            Array [
              280,
              "Students",
            ]
          }
        />
        <StatisticsNumbers
          course={
            Array [
              125,
              "Open and continuing training courses
",
            ]
          }
          students={
            Array [
              111,
              "Students",
            ]
          }
        />
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-end",
            }
          }
        >
          <Styled(Icon)
            name="chevron-down"
            style={
              Object {
                "color": "white",
              }
            }
            type="FontAwesome"
          />
        </TouchableOpacity>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Content)>
  <Styled(Content)
    contentContainerStyle={
      Object {
        "flexDirection": "column",
        "height": 1254,
        "justifyContent": "center",
        "padding": 0,
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "flex": 9,
          "flexDirection": "column",
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-start",
            }
          }
        >
          <Styled(Icon)
            name="chevron-up"
            type="FontAwesome"
          />
        </TouchableOpacity>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "backgroundColor": "white",
            "flex": 15,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-evenly",
            "padding": 40,
            "paddingTop": 0,
          }
        }
      >
        <StatisticsNumbers
          icon="flask"
          iconsNmb={18}
          percentage={90}
          text="Teachers and Reseachers

"
        />
        <StatisticsNumbers
          icon="globe"
          iconsNmb={18}
          percentage={6}
          text="Foreign students in mobility programs

"
        />
        <StatisticsNumbers
          icon="book"
          iconsNmb={18}
          percentage={18}
          text="Foreign students enrolled to obtain a degree
"
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1c1c1c",
          "flex": 8,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <Styled(ViewNB)
        key="0"
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "flex": 1,
            "flexDirection": "column",
            "padding": "2%",
          }
        }
      >
        <Fact
          info="240 protocolos estágio com empresas"
          key="0"
          name="feup"
        />
        <Fact
          info="30 contratos projetos prestação de serviços"
          key="1"
          name="feup"
        />
      </Styled(ViewNB)>
      <Styled(ViewNB)
        key="2"
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "flex": 1,
            "flexDirection": "column",
            "padding": "2%",
          }
        }
      >
        <Fact
          info="17 protocolos com escolas de ensino básico e secundário"
          key="0"
          name="feup"
        />
        <Fact
          info="30 trabalhos de tradução e revisão jurídica"
          key="1"
          name="feup"
        />
      </Styled(ViewNB)>
      <Styled(ViewNB)
        key="4"
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "flex": 1,
            "flexDirection": "column",
            "padding": "2%",
          }
        }
      >
        <Fact
          info="8 campanhas de angariação de bens"
          key="0"
          name="feup"
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Content)>
</ScrollView>
`;
